pkg/monitor/pod.go

```diff
@@ -619,3 +628,75 @@ func conditionsForTransitioningContainer(pod *corev1.Pod, current, previous *cor
 func isMirrorPod(pod *corev1.Pod) bool {
 	return len(pod.Annotations["kubernetes.io/config.mirror"]) > 0
 }
+
+type podNetworkIPCache struct {
+	lock sync.Mutex
+
+	// podIPsToCurrentPodLocators contains the name of the pods currently using a given pod IP.
+	// This only tracks the current state, because we will emit monitor error events on any overlaps.
+	podIPsToCurrentPodLocators map[string]sets.String
+}
+
+func (p *podNetworkIPCache) updatePod(pod, _ *corev1.Pod) []monitorapi.Condition {
+	var conditions []monitorapi.Condition
+
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	// only consider pod network pods because host network pods will have duplicated IPs
+	if pod.Spec.HostNetwork {
+		return conditions
+	}
+
+	podLocator := monitorapi.LocatePod(pod)
+
+	if pod.DeletionTimestamp != nil {
```

Comment:
if a pod is completed you also need to remove it from your tracker. There can be duplicate IPs on pods that are in completed state.

Reply:
> if a pod is completed you also need to remove it from your tracker. There can be duplicate IPs on pods that are in completed state.

Please see five lines down in this method.

Reply:
I think I found it.  The .status.Conditions[name=PodCompleted]